Sources/LiveViewNative/Coordinators/LiveSessionCoordinator.swift

```diff
@@ -364,19 +364,21 @@ public class LiveSessionCoordinator<R: RootRegistry>: ObservableObject {
     func redirect(_ redirect: LiveRedirect) async throws {
         switch redirect.mode {
         case .replaceTop:
-            await navigationPath.last?.coordinator.disconnect()
-            let coordinator = LiveViewCoordinator(session: self, url: redirect.to)
-            let entry = LiveNavigationEntry(url: redirect.to, coordinator: coordinator)
             switch redirect.kind {
             case .push:
+                let coordinator = LiveViewCoordinator(session: self, url: redirect.to)
+                let entry = LiveNavigationEntry(url: redirect.to, coordinator: coordinator)
                 navigationPath.append(entry)
             case .replace:
                 // If there is nothing to replace, change the root URL.
                 if !navigationPath.isEmpty {
-                    navigationPath[navigationPath.count - 1] = entry
+                    await navigationPath.last?.coordinator.disconnect()
+                    let coordinator = LiveViewCoordinator(session: self, url: redirect.to)
+                    coordinator.url = redirect.to
+                    navigationPath[navigationPath.count - 1] = LiveNavigationEntry(url: redirect.to, coordinator: coordinator)
+                    try await coordinator.connect(domValues: self.domValues, redirect: true)
                 }
             }
-            try await coordinator.connect(domValues: self.domValues, redirect: true)
```

Comment:
Connection is now handled from `scan` in `LiveSessionCoordinator.init`.

Only `replace` navigation needs to manage its own reconnection.